docker/multi-process/Dockerfile

```diff
@@ -1,32 +1,24 @@
 FROM ubuntu:14.04
 
-ADD docker/scripts/prepare /scripts/prepare
+COPY docker/scripts/prepare /scripts/
 RUN /scripts/prepare
 
-ADD docker/multi-process/scripts/standalone-packages /scripts/standalone-packages
+COPY docker/multi-process/scripts/standalone-packages /scripts/
 RUN /scripts/standalone-packages
 
 WORKDIR /app
-
-ADD ["Gemfile", "Gemfile.lock", "/app/"]
-ADD lib/gemfile_helper.rb /app/lib/
-ADD vendor/gems /app/vendor/gems
-
-RUN chown -R huginn:huginn /app && \
-    sudo -u huginn -H echo "gem 'sqlite3', '~> 1.3.11'" >> /app/Gemfile && \
-    sudo -u huginn -H LC_ALL=en_US.UTF-8 RAILS_ENV=production ON_HEROKU=true bundle install --without test development --path vendor/bundle -j 4
```

Comment:
We install the required gems here to utilize layer caching when building the docker image. If just one or a few gem versions changed the build process is much quicker because the majority of gems are already installed in a previous layer. Does that not work with your changes?

Reply:
I realized that the snippet was probably put here to utilize layer caching. But if gem versions are changed, then Gemfile / Gemfile.lock will be different, thus invalidating the layer at `ADD ["Gemfile", "Gemfile.lock", "/app/"]` already. A whole new `gem install` will then always happen in this case. But you're right, I should have deleted the first, duplicated `bundle install` in the `/scripts/setup` instead; this way, if other files introduced by `COPY . /app` were changed, `bundle install` will not be repeated. I was experimenting if I can move the whole `RUN /scripts/setup` here, and ended up not restoring this correctly. I'll make the changes.

Reply:
Oh, yes you are right. It was about caching the gem bundle if only some other files changed.